### PR TITLE
Run Settings control Updates.

### PR DESCRIPTION
**Pull request by ikeough**
This pull request "fixes" two bugs noted by @riteshchandawar:

http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6469
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6471

In addition, this pull request enacts stylistic changes agreed upon in the final review of the new RunSettings control.

- [x] Hide the Run Settings control in custom nodes.
- [x] Make the debug button only available in debug mode.
- [x] Make all new files open in Run Automatically.
- [x] Make alll existing files without a RunType open in Run Manually.
- [x] Make all other files honor the RunType setting saved in the file.
- [x] Change RunTypes to adjectives.
- [x] Make the Run button only visible during "Manual", and we move it to the right of the RunType drop-down.
- [x] Ensure that the bottom bar warning messages are working.
- [x] Make the message persistent until the user clicks anywhere in Dynamo.
- [x] Ensure all RunSettings test continue to pass.

The RunTypes drop-down has been moved to the far left, and the RunType enumeration now uses adjectives instead of adverbs:
image

The Run button is now hidden in all modes but Manual mode where it now displays to the right of the RunTypes drop-down:
image

The warning_text object has been replaced by the NotificationsControl which now displays to the right of the RunSettings control. This control displays the most recent posted notification.
image

PTAL:
- [ ] @pboyer

### CODE REVIEW

File: src/DynamoCore/Core/Notifications.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+namespace Dynamo.Core
+{
+    public enum NotificationLevel { Mild, Moderate, Error }
+
+    /// <summary>
+    /// The Notifications class is used to raise notifications that
+    /// can be handled by the UI or the logger.
+    /// </summary>
+    public class Notifications
```
Comment (ikeough) [inline src/DynamoCore/Core/Notifications.cs]: The Notifications class is created to serve the specific purpose of raising notifications that can be handled anywhere in the system with a reference to DynamoCore.

Notifications should ultimately replace the use of the Logger for bubbling warnings up to the UI. The Logger should be exclusively for writing to the log. This is outside of the scope of this PR, but is covered in [this internal task](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6549).

File: src/DynamoCoreWpf/Controls/NotificationsControl.xaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+﻿<UserControl x:Class="Dynamo.Wpf.Controls.NotificationsControl"
```
Comment (ikeough) [inline src/DynamoCoreWpf/Controls/NotificationsControl.xaml]: The NotificationsControl replaces the "warning_text" object in the DynamoView. This control instantiates a NotificationsViewModel object to use as its DataContext. The NofificationsViewModel listens for NotificationPosted events on the Notifications singleton and store the message and the warning level on two public properties bound to the UI.

File: src/DynamoCoreWpf/Controls/NotificationsControl.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        /// <summary>
+        /// Handle the DynamoView's PreviewMouseDown event.
+        /// When the user click anywhere in the view, clear the ViewModel's warning.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        void window_PreviewMouseDown(object sender, System.Windows.Input.MouseButtonEventArgs e)
+        {
```
Comment (ikeough) [inline src/DynamoCoreWpf/Controls/NotificationsControl.xaml.cs]: The current notification is cleared when the user clicks anywhere else in the DynamoView.

File: src/DynamoCoreWpf/Controls/RunSettingsControl.xaml [hunk 4]
```diff
@@ -66,35 +54,57 @@
...
                     Text="{Binding RunPeriod, Converter={StaticResource RunPeriodConverter}, UpdateSourceTrigger=Explicit}"
                      VerticalContentAlignment="Center"
                      Visibility="{Binding RunPeriodInputVisibility}"
                      Width="80" FontSize="14px"
-                     Style="{DynamicResource ResourceKey=SDarkTextBox}" Foreground="#bbbbbb" 
+                     Style="{DynamicResource ResourceKey=SDarkTextBox}" 
+                     Foreground="#bbbbbb" 
                      KeyDown="UIElement_OnKeyDown" Margin="2.5,0,0,0"/>
 
-            <!--<Button Name="cancelButton"
```
Comment (ikeough) [inline src/DynamoCoreWpf/Controls/RunSettingsControl.xaml]: The cancel button is removed entirely as it was never wired to anything.

Comment (pboyer) [inline src/DynamoCoreWpf/Controls/RunSettingsControl.xaml]: Really?  I thought @aparajit-pratap did a good amount of work to allow a cancellation.

File: src/DynamoCore/Core/Notifications.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        /// <summary>
+        /// The Notification's message;
+        /// </summary>
+        public string Message { get; internal set; }
+
+        /// <summary>
+        /// The level of the Notification.
+        /// </summary>
+        public NotificationLevel Level { get; internal set; }
+
```
Comment (pboyer) [inline src/DynamoCore/Core/Notifications.cs]: Hmm.. This assumes we have only a single instance of the Dynamo application.  Is that a good assumption?  Maybe each Dynamo session needs a unique identifier.

File: src/DynamoCore/Models/HomeWorkspaceModel.cs [hunk 6]
```diff
@@ -325,6 +325,7 @@ private void OnUpdateGraphCompleted(AsyncTask task)
 
             // Notify listeners (optional) of completion.
             RunSettings.RunEnabled = true; // Re-enable 'Run' button.
+            Notifications.Instance.PostNotification("Run complete.", NotificationLevel.Mild);
```
Comment (pboyer) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: What if there are multiple home workspaces?  The notification needs to somehow tell the callee about which workspace.

File: src/DynamoCore/Core/Notifications.cs [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+            if (NotificationPosted != null)
+            {
+                NotificationPosted(args);
+            }
+        }
+
+        private Notifications(){}
+
+        public void PostNotification(string message, NotificationLevel level)
+        {
```
Comment (pboyer) [inline src/DynamoCore/Core/Notifications.cs]: How about `Post`?

File: src/DynamoCore/Core/Notifications.cs (hunk 1 above)
Comment (pboyer) [inline src/DynamoCore/Core/Notifications.cs]: How about `Notifier`?

File: src/DynamoCoreWpf/Controls/RunSettingsControl.xaml (hunk 4 above)
Comment (ikeough) [inline src/DynamoCoreWpf/Controls/RunSettingsControl.xaml]: I'll take another look, but all reports are that pressing the cancel button didn't do diddly.

Comment (ikeough) [inline src/DynamoCoreWpf/Controls/RunSettingsControl.xaml]: The Cancel button was wired to the RunCancelCommand, which calls the Run method on HomeWorkspaceModel. It doesn't actually cancel anything.

Comment (ikeough) [inline src/DynamoCoreWpf/Controls/RunSettingsControl.xaml]: I think @Benglin can shed some light here. Looks like cancellation functionality was removed when we switched over to the scheduler, and a task was created to put it back, but we never got around to it.

File: src/DynamoCoreWpf/ViewModels/Core/HomeWorkspaceViewModel.cs [hunk 8]
```diff
@@ -26,6 +58,40 @@ public HomeWorkspaceViewModel(HomeWorkspaceModel model, DynamoViewModel dynamoVi
...
+            else
+            {
+                SetCurrentWarning(NotificationLevel.Moderate, "Run completed with warnings."); 
+            }
+        }
+
+        void hwm_EvaluationStarted(object sender, EventArgs e)
+        {
+            SetCurrentWarning(NotificationLevel.Mild, "Run completed...");
+        }
```
Comment (pboyer) [inline src/DynamoCoreWpf/ViewModels/Core/HomeWorkspaceViewModel.cs]: You mean started?

File: src/DynamoCoreWpf/ViewModels/Core/HomeWorkspaceViewModel.cs [hunk 9]
```diff
@@ -26,6 +58,40 @@ public HomeWorkspaceViewModel(HomeWorkspaceModel model, DynamoViewModel dynamoVi
...
+            }
+            else
+            {
+                SetCurrentWarning(NotificationLevel.Moderate, "Run completed with warnings."); 
+            }
+        }
+
+        void hwm_EvaluationStarted(object sender, EventArgs e)
+        {
+            SetCurrentWarning(NotificationLevel.Mild, "Run started...");
```
Comment (pboyer) [inline src/DynamoCoreWpf/ViewModels/Core/HomeWorkspaceViewModel.cs]: PLEASE LOCALIZE THX!

**Comment by pboyer** [comment]
Give me localization or give me death!  Then LGTM!

**Pull request merged**